### DISCUSSION TITLE
Long Network response time

### DISCUSSION THREAD

**Discussion by combaudouo** [leokhoa/laragon#415] [Q&A]
Hi,

I have been using Laragon for quite a long time now and network response time is very long when I refresh a page (between 5s and 13s). I have tried a lot of things (changing the database host from localhost to 127.0.0.1, removing some chunks of the code, ...) but I am a bit stuck and it's really painful...

I have tried to start a fresh laravel install, or a fresh wordpress install, but it does the same. However, when I start a blank project, I have no network issue (even when I try to make a mysql connection). So I imagine there may be something with laragon config or something...

Anyone has already experienced that ?

<img width="400" alt="image" src="https://user-images.githubusercontent.com/5708504/220587023-0bcd094d-bc97-4294-97af-746f7517ed5e.png">

Thanks a lot,

**Comment by Pen-y-Fan** [comment]
Yes, I've had the same on long-running projects. I have never found the course. The task manager doesn't show any bottlenecks.

I have a long-running project with 800 tests, running an in-memory SQLite database.

* my 5-year-old Windows 10 work laptop running Laragon takes 4 minutes to run the tests. 
* my 1-year-old personal windows 11 laptop running Laragon takes 1 minute! Exactly the same code, similar spec i7 laptops. 
* I've been given a 5-year-old Mac book Pro, running Valet, to try and it takes 1 minute.

It's not only a network response time. It's fine for a new project on Laragon. At least on newer, different OS and hardware.

This is one of the reasons I do a clean install, rather than an upgrade of Laragon (see [How to update? #405](https://github.com/leokhoa/laragon/discussions/405)), the "new install" appears to fly, for a short time.

My colleagues have also complained their Laptops are slow. 

We are due a hardware refresh:

* I've got a new MacBook M2 on order 🎉.  
* my colleagues have a new Dell XPS on order, which will stay with Laragon, so we may see if it's just an "old hardware" issue.

Even though I 💖 Laragon, I can't figure out the problem. Why does exactly the same code run x4 slower on my old Windows laptop? Is it a slowing of the SSD? Anti-virus? Big log file? MySQL log? Conflict of some sort? 🤷

It doesn't seem to be a Laragon problem, as it runs fine on my personal laptop. I will be staying with Lagon on that. My personal projects are usually short-lived, small databases.

**Reply by Pen-y-Fan** [reply]
I had my M2 Macbook today:

- my 5-year-old Windows 10 work laptop running Laragon takes 4 minutes to run the tests. 💩
- my 1-year-old personal windows 11 laptop running Laragon takes 1 minute! Exactly the same code, similar spec i7 laptops.
- I've been given a 5-year-old Mac book Pro intel i7, running Valet, to try and it takes 1 minute.
- My colleague ran the tests using his 1-year-old Mac book Pro M1, running Valet, and it takes 27 seconds
- My new M2 Mac book pro m2 running Valet 15 seconds 🎉

**Comment by cistelsa** [comment]
Excuse me for my English, but I gave a solution to a slow response from the network, more than anything associated with the connection to the MySQL database, I updated it to MaríaDB and these are the steps to follow:

1. Download the version of maridb in my case Version 10.9.5 (I think it is the most stable) download in the [link](https://mariadb.org/download/?t=mariadb&p=mariadb&r=10.9.5&os=windows&cpu=x86_64&pkg=zip&m=insacom)
2. unpack it and upload it to the /bin/mysql/ folder
3. activate the version in laragon
4. open the console and write the following command mysql_secure_installation follow the steps described including the root password, this is to start mariadb since it does not have the password and several startup parameters configured.
5. You can now start creating, importing, and modifying the root password from laragon and everything works much faster.

You can comment if this solution turned out to be perfect for me since it was taking up to 3 seconds to load anything simple, with this it was in less than 1 second.

**Reply by combaudouo** [reply]
Thanks for your help. I just tried it, and I see no improvement on my side, so I imagine it comes from somewhere else

**Comment by combaudouo** [comment]
Clearly an issue with Windows... I tried to deploy my code on a cloud hosting, it worked like a charm

**Comment by mohamedds-12** [comment]
Having the same problem, I can't find a solution to it. Note that when I do `php artisan serve` it works fast so I don't think it's an issue with windows like @combaudouo suggested.

**Comment by prathamesh-gharat** [comment]
Check if it takes the same time via `curl` compared to when you load it via the browser. In my case I was experiencing a 1s delay even with a simple file containing only `phpinfo();` too. It turned out to be a browser related issue where the browser was trying to load favicon.ico, which was missing and since it was missing the request was ending upon a WordPress 404 page (which was taking 1 second).

Note: It was the browser that was `blocking` (as in putting other requests on hold until it was done) the network request in my case, not the server. (tested by keeping the server busy with apache bench and requesting another URL on the same sub-domain from the browser)

**Reply by prathamesh-gharat** [reply]
I am not sure if my observation is specific to my system or Windows in general.
I am on Windows 10.

I had a WordPress website with DB_HOST configured as 'localhost'
Switching to '127.0.0.1' shaved off around 1.5 second. Really odd.

P.S.: on my system, localhost is bound to IPv6 loopback `::1`. Not sure if/how that matters. My local MySQL is set to listen on all interfaces `0.0.0.0` (so that I can access it from a system on Docker)

**Comment by josealgardataborges** [comment]
Don't get me wrong, but browsing speed on local development servers, is **always** related to hardware/system speed, it really doesn't have much to do with LARAGON.

You can point your finger at apache/mysql/php configurations, but not at LARAGON.

Check your "applications" (apache/mysql/php) settings and see where it can be improved... For instance if you have 16GB of RAM, increase the RAM available to mysql for instance... If you have a 12core CPU increase the number of PHP processes, etc.

But, as mentioned, and in my opinion, **this has nothing to do with LARAGON per say**.

**Reply by mohamedds-12** [reply]
I think it's related to Laragon because when I do `php artisan serve` it works fine and fast, So I don't think it's related to apache's config or MySQL's or PHP's, at least for my case.

**Reply by prathamesh-gharat** [reply]
Laragon is just helping us manage the different servers, so it wouldn't affect performance directly. It might be related to how Laragon runs/configures (default) apache & PHP. I will be debugging my current performance issue with qCacheGrind only appears on my local PC and not on a ubuntu server. If I find something, I'll update you guys.

P.S. I just tried disabling Apache entirely and Enabling Nginx on 80/443 .. there was no change for my case. I am suspecting there is some CURL call happening in my WP instance which takes longer from my local PC 🤔 .. I will confirm after I have traced the issue with xdebug.

**Reply by josealgardataborges** [reply]
Laragon doesn't change apache, php or mysql configuration files. it just "manages" the service itself.

It's like saying "the car is not fast, because of the leather seats." ;)

LARAGON is not intercepting the HTTP requests (apache is) or connections do BD (Mysql is).

You just unpack the default apache, mysql, php zip files, with default settings, not optimized for your machine, and expect it to run smooth and fast.

You have to tweek those settings, to your needs.

Sure "LARAGON" can change some settings IF you tell him to, but by default, it doesn't.

**Comment by leokhoa** [comment]
@mohamedds-12 : Can you provide further information:
1. How fast when running `php artisan serve` compare to Apache **"with the same port"**?
2. How fast when running `php artisan serve` compare to Nginx **"with the same port"**?

**Comment by josealgardataborges** [comment]
`php artisan serve` uses **PHP internal server**, it's the same thing as running `php -S localhost:80`

It does not use Apache!

**Comment by combaudouo** [comment]
I suspect on my side an issue with the default conf of mysql. With blank projects, there is no slowdown. I have tried to use mysql in a blank project and a simple query, it worked but maybe with more complex queries...

**Reply by prathamesh-gharat** [reply]
As a project advances in time, you are also adding in new packages. So it can be the overhead of packages + the DB. You can easily profile the SQL queries as you are using Laravel by enabling debug mode (APP_DEBUG).

P.S. Look into opcache if you want to speed up local dev env a bit, but understand how the cache works otherwise it might get annoying for you while development.

**Reply by combaudouo** [reply]
I have the same issue with a default laravel install or a default symfony install. 

I have telescope installed on my project => 112ms for 42 queries, so it's acceptable

I have enabled opcache, but no dramatic change so far

**Comment by Andres-M-Rios** [comment]
Hello, 

 

Interesting observation, but not ODD at all. In fact this is an artefact of DNS resolution.

 

Any time DNS resolution is required you are adding seconds to any transaction.

 

Unfortunately, with any production site, you will need to use DNS.

 

With a home DEV setup you can cheat a little bit. 127.0.0.1 is an OLD low level IP HACK that is built into any modern-day TCP/IP v4 compatible setup.

 

You specify the IP, you save DNS, and you shave seconds off your process.

 

It’s not odd, it’s just how it works.

 

Just like a V6 is faster than a V4 engine.

 

++++

Andres M. Partridge-Rios

**Comment by Fresh121212** [comment]
I was puzzled by this issue testing a Drupal install locally on Laragon 6. A page takes forever (minutes) to load. I tried it on a "Private Browsing" instance of FF and it was loaded in a blink of an eye. I'm guessing it has to do with browser/cache!

**Reply by Pen-y-Fan** [reply]
This could be an extension, try another browser (Edge, Firefox developer edition) without any installed extensions.

My colleague has a problem with Chrome, which has extensions such as add blocker installed (which is the suspected problem), when using the debug bar and Chrome developer tools the local site grinds to a halt. Using another browser, without extensions, fixed the problem.

**Reply by Fresh121212** [reply]
Thank you, yes like I said, I tried it in a "Private Browsing" instance of Mozilla (no addons/extensions) and it's been working well. I doubt it's an extension thing in my case as it used to work well before with same extensions.

**Comment by cvrle77** [comment]
If I ever come back to this, issue is misconfigured DNS. No idea how, where or why, however, I found a solution.
Go to Tools/WildcardDNS, switch to nip.io, then to xip.io, then disable. Site speed should be restored to blazing fast.
This thread was one of few I was looking for solution, but no one offered anything, so I had to be creative and move everything around to find the solution.

**Reply by bagongd3** [reply]
Thaks bro.. it's work for me

**Reply by praveenkalaiarasu** [reply]
Worked for me too. 
I my high spec-ed laptop I never laragon to be slow. 
But in my secondary laptop it has been always slow.
This fixed it. 

I tried leaving it in nip.io itself. That seems to be a even a little bit more responsive than switching back to disabled. Might be placebo effect.

**Reply by TimothyRT** [reply]
Thank you; that works for me as well!
